x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -173,44 +179,49 @@ export const signalRulesAlertType = ({
-      let wroteStatus = false;
+      let wrotePartialFailureStatus = false;
       await ruleStatusService.goingToRun();
 
       // check if rule has permissions to access given index pattern
       // move this collection of lines into a function in utils
       // so that we can use it in create rules route, bulk, etc.
       try {
-        const inputIndex = await getInputIndex(services, version, index);
-        const privileges = await checkPrivileges(services, inputIndex);
-
-        const indexNames = Object.keys(privileges.index);
-        const [indexesWithReadPrivileges, indexesWithNoReadPrivileges] = partition(
-          indexNames,
-          (indexName) => privileges.index[indexName].read
-        );
+        const [privileges, timestampFieldCaps] = await Promise.all([
+          flow(
+            () => tryCatch(() => getInputIndex(services, version, index), toError),
+            chain((indices) => tryCatch(() => checkPrivileges(services, indices), toError)),
+            flow(toPromise)
+          )(),
+          services.scopedClusterClient.fieldCaps({
+            index,
+            fields: timestampOverride != null ? ['@timestamp', timestampOverride] : ['@timestamp'],
+            allow_no_indices: false,
+            include_unmapped: true,
+          }),
+        ]);
 
-        if (indexesWithReadPrivileges.length > 0 && indexesWithNoReadPrivileges.length > 0) {
-          // some indices have read privileges others do not.
-          // set a partial failure status
-          const errorString = `Missing required read permissions on indexes: ${JSON.stringify(
-            indexesWithNoReadPrivileges
-          )}`;
-          logger.debug(buildRuleMessage(errorString));
-          await ruleStatusService.partialFailure(errorString);
-          wroteStatus = true;
-        } else if (
-          indexesWithReadPrivileges.length === 0 &&
-          indexesWithNoReadPrivileges.length === indexNames.length
-        ) {
-          // none of the indices had read privileges so set the status to failed
-          // since we can't search on any indices we do not have read privileges on
-          const errorString = `The rule does not have read privileges to any of the following indices: ${JSON.stringify(
-            indexesWithNoReadPrivileges
-          )}`;
-          logger.debug(buildRuleMessage(errorString));
-          await ruleStatusService.error(errorString);
-          wroteStatus = true;
-        }
+        wrotePartialFailureStatus = await flow(
+          () =>
+            tryCatch(
+              () => hasReadIndexPrivileges(privileges, logger, buildRuleMessage, ruleStatusService),
+              toError
+            ),
+          chain((wroteStatus) =>
+            tryCatch(
+              () =>
+                hasTimestampFields(
+                  wroteStatus,
+                  params.timestampOverride != null ? params.timestampOverride : '@timestamp',
```

Comment:
Can this be:
```suggestion
                  params.timestampOverride ?? '@timestamp',
```

Reply:
Ah actually this should have been using `isEmpty` but yes that would have been cleaner too. Sorry 😬